tests/fast-integration/eventing-test/assets/eventing-function.yaml

```diff
@@ -0,0 +1,132 @@
+        function sendEvent(url, headers, payload) {
+            console.log("Forwarding event to EPP: " + url)
+            console.log("Payload: ", payload)
+            console.log("Headers: ", headers)
+            return axios.post(url, payload, {headers:headers})
+            return response
+        }
+        function cloudEventHeaders(event) {
+          return Object.keys(event).reduce((accumulator, current) => {
+            if (current.startsWith('ce-')) {
+              accumulator[current] = event[current]
+            }
+            return accumulator;
+          }, {});
+        }
+        function dump(obj) {
+          console.dir(obj, {depth: null});
+        }
+        module.exports = {
+          main: async function (event, context) {
+            var metadata = { podName: process.env.HOSTNAME }
+
+            if (event.extensions.request.query.send) {
+              // Send a request with param send=true to publish event to EPP.
+              try {
+                const reqBody = event.data
+                const response = await sendEvent(reqBody.url, reqBody.data.headers, reqBody.data.payload);
+                console.log(`In-app event sent. Publisher proxy response: ${response.status}`);
+                return { success: true, status: response.status, metadata: metadata }
+              } catch (e) {
+                console.dir(e);
+                return { success: false, errorMessage: e.message, error: JSON.stringify(e), metadata: metadata }
+              }
+            } else if (event.extensions.request.query.eventid) {
+              // Send a request with param eventid=<id> to fetch an received event.
+              console.log("Searching and returing event with id: ", event.extensions.request.query.eventid);
+              result = receviedEvents[event.extensions.request.query.eventid]
+              if (result) {
+                console.log("event found with id: ", event.extensions.request.query.eventid);
+                return { success: true, event:result, metadata: metadata};
+              }
+              console.log("event NOT found with id: ", event.extensions.request.query.eventid);
+              return { success: false, metadata: metadata};
+            } else if (event["ce-type"]){
+              // Store all valid received events.
+              console.log("Got in-app event:", event["ce-type"], ", id: ", event["ce-id"]);
+              savedFormat = { ceHeaders: cloudEventHeaders(event), payload: event.data, headers: event.extensions.request.headers }
+              receviedEvents[event["ce-id"]] = savedFormat;
+              return { success: true }
+            } else {
+              console.log("Discarding event with ce-type:", event["ce-type"])
+              console.log("Discarding event with data:", event.data)
+            }
+            return {metadata: metadata};
+          }
+        }
+---
+apiVersion: hydra.ory.sh/v1alpha1
+kind: OAuth2Client
```

Comment:
why do we need this? we only use in-cluster eventing, so a full blown oauth2 workflow seems unnecessary in a test

Reply:
I have removed auth for this.